src/widget.ts

```diff
@@ -105,29 +104,59 @@ export class BabylonPointCloudModel extends BabylonBaseModel {
 }
 
 export class BabylonPointCloudView extends BabylonBaseView {
-
   protected async createScene(): Promise<Scene> {
-    return super.createScene().then( ( scene ) => {
-      const data = this.values.data;
+    return super.createScene().then(async scene => {
+
+      var isTime = false;
+      var isClass = false;
+      var isTopo = false;
+      var isGltf = false;
+
+      var data!: { X: number[], Y: number[], Z: number[], Red: number[], Green: number[], Blue: number[]}; 
+
+      console.log(this.values);
+
+      if (this.values.mode === "time"){
+        isTime = true;
+      }else if (this.values.mode === "classes") {
+        isClass = true;
+      }else if(this.values.mode == "topo"){
+        isTopo = true;
+      }else if(this.values.mode == "gltf"){
+        isGltf = true;
+      } 
+      
+      console.log(this.values.source);
```

Comment:
debug - can you remove all the console.log statements unless they are actual errors.